src/main/java/com/ibm/devops/connect/CRPipeline/UploadASoCTestResult.java

```diff
@@ -0,0 +1,284 @@
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.StaplerRequest;
+
+import java.util.Map;
+
+import org.apache.http.entity.mime.MultipartEntityBuilder;
+import org.apache.http.HttpEntity;
+import org.apache.http.entity.ContentType;
+
+import com.ibm.devops.connect.CloudPublisher;
+
+public class UploadASoCTestResult extends Notifier {
+
+    private String tenantId;
+    private String environment;
+    private String appId;
+    private String appExtId;
+    private String appName;
+    private String buildUrl;
+    private String metricDefinition;
+    private String recordName;
+    private String commitId;
+
+    @DataBoundConstructor
+    public UploadASoCTestResult(
+        String tenantId,
+        String environment,
+        String appId,
+        String appExtId,
+        String appName,
+        String buildUrl,
+        String metricDefinition,
+        String recordName,
+        String commitId
+    ) {
+        this.tenantId = tenantId;
+        this.environment = environment;
+        this.appId = appId;
+        this.appExtId = appExtId;
+        this.appName = appName;
+        this.buildUrl = buildUrl;
+        this.metricDefinition = metricDefinition;
+        this.recordName = recordName;
+        this.commitId = commitId;
+    }
+
+    public String getTenantId() { return tenantId; }
+    public String getEnvironment() { return environment; }
+    public String getAppId() { return appId; }
+    public String getAppExtId() { return appExtId; }
+    public String getAppName() { return appName; }
+    public String getBuildUrl() { return buildUrl; }
+    public String getMetricDefinition() { return metricDefinition; }
+    public String getRecordName() { return recordName; }
+    public String getCommitId() { return commitId; }
+    
+    @Override
+    public BuildStepMonitor getRequiredMonitorService() {
+        // BUILD means this step will only be run after the previous build is
+        // fully completed
+        return BuildStepMonitor.BUILD;
+    }
+
+    @Override
+    public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener)
+            throws AbortException, InterruptedException, IOException {
+
+        EnvVars envVars = build.getEnvironment(listener);
+        // Resolving all passed ${VARIABLES}
+        String tenantIdValue = envVars.expand(this.tenantId);
+        String environmentValue = envVars.expand(this.environment);
+        String appIdValue = envVars.expand(this.appId);
+        String appExtIdValue = envVars.expand(this.appExtId);
+        String appNameValue = envVars.expand(this.appName);
+        String buildUrlValue = envVars.expand(this.buildUrl);
+        String metricDefinitionValue = envVars.expand(this.metricDefinition);
+        String recordNameValue = envVars.expand(this.recordName);
+
+        try {
+            // thread to sleep for 1000 milliseconds
+            Thread.sleep(60000);
```

Comment:
What is the purpose of this sleep?  Also, the comment on the line above does not match